### PR TITLE
{Core} Disable aladdin recommendation in autocomplete mode

### PR DESCRIPTION
**Pull request by bebound**
**Description**

If the command is not complete, you will get recommendation.
For example, type `az account` you get
```                                                                                                                                                ─╯
the following arguments are required: _subcommand

Examples from AI knowledge base:
az account list
Get a list of subscriptions for the logged in account. (autogenerated)

az account show
Get the details of a subscription. (autogenerated)

az account set --subscription mysubscription
Set a subscription to be the current active subscription. (autogenerated)

https://docs.microsoft.com/en-US/cli/azure/account#az_account_list
Read more about the command in reference docs
```
In autocomplete mode, it's common that the command is not complete. The recommendation is useless and time-consuming.
So I disable it in autocomplete mode.

**Testing Guide**

`time IFS=$'\013' _ARC_DEBUG=1 COMP_LINE='az account' COMP_POINT=11 COMP_TYPE="" _ARGCOMPLETE=1 _ARGCOMPLETE_SUPPRESS_SPACE=0 az 8>&1 1>&1 2>&1 9>&1 ` takes 0.4s.
After apply this pr, it only takes 0.2s.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Core

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/util.py [hunk 1]
```diff
@@ -1360,3 +1360,9 @@ def should_encrypt_token_cache(cli_ctx):
     encrypt = cli_ctx.config.getboolean('core', 'encrypt_token_cache', fallback=fallback)
 
     return encrypt
+
+
+def is_autocomplete():
+    """Detect if the command is running in autocomplete mode"""
+    from knack.completion import ARGCOMPLETE_ENV_NAME
+    return os.environ.get(ARGCOMPLETE_ENV_NAME) is not None
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/util.py:1368]: I think we already have similar logic: https://github.com/Azure/azure-cli/blob/5c57d1c88bb5a8daded21e5fb6dc933456e9026d/src/azure-cli-core/azure/cli/core/__init__.py#L74

File: src/azure-cli-core/azure/cli/core/command_recommender.py [hunk 2]
```diff
@@ -357,6 +359,9 @@ def _disable_aladdin_service(self):
         if self.cli_ctx.__class__.__name__ == 'DummyCli':
             return True
 
+        if is_autocomplete():
+            return True
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/command_recommender.py:363]: This can be retrieved from `self.cli_ctx.data['completer_active']`.

Comment (bebound) [reply src/azure-cli-core/azure/cli/core/command_recommender.py]: Nice catch, I'll update #24765 too.

**Review by jiasli — APPROVED** [review]

**Pull request merged**